eZ/Bundle/EzPublishRestBundle/Tests/Functional/RoleTest.php

```diff
@@ -633,4 +676,52 @@ private function roleDraftHrefToRoleHref($roleDraftHref)
     {
         return str_replace('/draft', '', $roleDraftHref);
     }
+
+    /**
+     * Creates and published a role with $identifier.
+     *
+     * @param string $identifier
+     *
+     * @return string The href of the published role
+     */
+    private function createAndPublishRole($identifier)
+    {
+        $xml = <<< XML
+<?xml version="1.0" encoding="UTF-8"?>
+<RoleInput>
+  <identifier>$identifier</identifier>
+  <mainLanguageCode>eng-GB</mainLanguageCode>
+  <names>
+    <value languageCode="eng-GB">$identifier</value>
+  </names>
+  <descriptions>
+    <value languageCode="eng-GB">$identifier description</value>
+  </descriptions>
+</RoleInput>
+XML;
+        $request = $this->createHttpRequest(
+            'POST',
+            '/api/ezp/v2/user/roles?publish=false',
+            'RoleInput+xml',
+            'RoleDraft+json'
+        );
+        $request->setContent($xml);
+        $response = $this->sendHttpRequest($request);
+
+        self::assertHttpResponseCodeEquals($response, 201);
+        self::assertHttpResponseHasHeader($response, 'Location');
+
+        $href = $response->getHeader('Location');
+        $this->addCreatedElement($href);
+
+        $roleDraftHref = $href . '/draft';
+
+        $response = $this->sendHttpRequest(
+            $this->createHttpRequest('PUBLISH', $roleDraftHref)
+        );
+
+        self::assertHttpResponseCodeEquals($response, 201);
```

Comment:
This will fail once https://github.com/ezsystems/ezpublish-kernel/pull/1628 is merged. Would it be evil to withhold my approval until then? :trollface: